loopchain/channel/channel_service.py

```diff
@@ -652,7 +649,9 @@ def __ready_to_height_sync(self):
         blockchain = self.block_manager.get_blockchain()
 
         blockchain.init_blockchain()
-        if blockchain.block_height == -1 and 'genesis_data_path' in conf.CHANNEL_OPTION[ChannelProperty().name]:
+        if blockchain.block_height == -1 \
+                and 'genesis_data_path' in conf.CHANNEL_OPTION[ChannelProperty().name] \
+                and self.is_support_node_function(conf.NodeFunction.Vote):
```

Comment:
if condition statement is too long, how about create check method that return `bool`.
for example,
```py
def _is_genesis(self):
    if condition1 and condtion2 and condtion3:
        return true
    return false
...
in another method
...

    if self._is_genesis():
        generate_genesis_block()
```

Reply:
in addition to, I suggest use parenthesis with multiline conditions.
```py
if (condition1
       and condition2
       and condition3):
    # do something..
```